### PR TITLE
feat(aapd-1195): move database to shared package

### PR DESCRIPTION
**Pull request by bryn-pins**
## Ticket Number

https://pins-ds.atlassian.net/browse/AAPD-1195

## Description of change

Moving prisma database logic to shared package so it can be reused in multiple packages
Leaving repo logic within the app itself for now
Pipeline now does migration/seeding in a job before api
Added the means to skip running tests in pipeline when running a manual build for speeding up dev workflow on pipelines

## Checklist

- [ ] Requires infrastructure changes
- [ ] I have updated the documentation accordingly
- [x] My commit history in this PR is linear
- [ ] New features have tests
- [ ] Breaking change (team conversation required)

## Important

Please do not merge from `main` (please only [rebase](https://github.com/Planning-Inspectorate/appeal-planning-decision/wiki/An-intro-to-Git-Rebase)). This keeps the history linear and easier to debug.

### CODE REVIEW

File: packages/database/.env.example
Comment (beejjacobs) [inline packages/database/.env.example:1]: Don't we still need the api one? Should this be a copy?

**Review by beejjacobs — COMMENTED** [review]
Was wondering if we'd need to update the api `index.d.ts`, but I think it'll still work. 

Looks good, a few thoughts to tidy things up.

File: packages/database/package.json [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"prisma": {
+		"schema": "src/schema.prisma",
+		"seed": "node src/seed/seed-dev.js"
+	},
+	"scripts": {
+		"db:migrate:dev": "npx prisma migrate dev",
+		"db:migrate:prod": "npx prisma migrate deploy",
+		"db:generate": "npx prisma generate",
+		"db:seed": "npx prisma db seed",
+		"db:seed:prod": "npm run db:generate && node src/seed/seed-prod.js",
```
[the remark this answers is not in the document]
Comment (bryn-pins) [reply packages/database/package.json:19]: have done this

File: packages/appeals-service-api/package.json [hunk 2]
```diff
@@ -8,15 +8,10 @@
 	},
 	"main": "src/main.js",
 	"prisma": {
-		"schema": "src/db/schema.prisma",
-		"seed": "node src/db/seed/seed-dev.js"
+		"schema": "../database/src/schema.prisma"
 	},
```
[the remark this answers is not in the document]
Comment (bryn-pins) [reply packages/appeals-service-api/package.json:12]: Couldn't manage to move the logic to database package in order to remove it although that might be possible, 
currently the schema is held in the shared package and that file is used to generate the client for the packages that need it

File: packages/appeals-service-api/package.json [hunk 3]
```diff
@@ -38,6 +33,7 @@
 		"@azure/service-bus": "^7.7.3",
 		"@pins/business-rules": "file:../business-rules",
 		"@pins/common": "file:../common",
+		"@pins/database": "file:../database",
 		"@prisma/client": "^5.6.0",
```
[the remark this answers is not in the document]
Comment (bryn-pins) [reply packages/appeals-service-api/package.json:37]: currently the client is being built for the api and it just pulls in the schema from database package

File: packages/appeals-service-api/Dockerfile [hunk 4]
```diff
@@ -15,11 +15,13 @@ COPY package.json /opt/
...
 COPY packages/common /opt/packages/common
 COPY packages/business-rules /opt/packages/business-rules
+COPY packages/database /opt/packages/database
 
 # install from root
 WORKDIR /opt
 RUN npm ci --omit=dev
 # generate prisma client
+WORKDIR /opt/packages/appeals-service-api
 RUN npm run db:generate
```
[the remark this answers is not in the document]
Comment (bryn-pins) [reply packages/appeals-service-api/Dockerfile:25]: if left as is, yes as the generate on root calls the one in database package and is used for seeding/migrations
individual packages will need their own generate command

File: packages/database/.env.example
Comment (bryn-pins) [reply packages/database/.env.example:1]: I think this is only used for seeding which has moved to database package
running locally will come from the docker compose env setting

**Review by beejjacobs — APPROVED** [review]

**Pull request merged**